TunnelVisionLabs.ReferenceAssemblyAnnotator/AssemblyResolver.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace TunnelVisionLabs.ReferenceAssemblyAnnotator
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.IO;
+    using Mono.Cecil;
+
+    internal sealed class AssemblyResolver : IAssemblyResolver
+    {
+        private static readonly ReaderParameters DefaultReaderParameters = new ReaderParameters(ReadingMode.Deferred);
+
+        private readonly string _searchDirectory;
+        private readonly ConcurrentDictionary<string, AssemblyDefinition?> _assembliesByName = new ConcurrentDictionary<string, AssemblyDefinition?>();
```

Comment:
I'm thinking it would be hugely beneficial to reuse this cache across all assemblies in parallel, even though the cache is currently recreated for each assembly and assemblies are currently annotated serially.

Reply:
On second thought, there's no way the object model can be accessed safely from concurrent threads when it is in deferred reading mode. Best we can do is to have a cache per thread.